src/control/lib/support/log.go

```diff
@@ -0,0 +1,640 @@
+		os.Rename(hiddenConf, nonhiddenConf)
+	}
+
+	return nil
+}
+
+// Collect all server side logs
+func CollectServerLog(log logging.Logger, opts ...Params) error {
+	var cfgPath string
+
+	if opts[0].Config != "" {
+		cfgPath = opts[0].Config
+	} else {
+		cfgPath, _ = getServerConf(log)
+	}
+	serverConfig := config.DefaultServer()
+	serverConfig.SetPath(cfgPath)
+	serverConfig.Load()
+
+	targetServerLogs, err := createHostLogFolder(serverLogs, log, opts...)
+	if err != nil {
+		return err
+	}
+
+	switch opts[0].LogCmd {
+	case "EngineLog":
+		for i := range serverConfig.Engines {
+			matches, _ := filepath.Glob(serverConfig.Engines[i].LogFile + "*")
+			for _, logfile := range matches {
+				err = cpLogFile(logfile, targetServerLogs, log)
+				if err != nil {
+					return err
+				}
+			}
+		}
+	case "ControlLog":
+		err = cpLogFile(serverConfig.ControlLogFile, targetServerLogs, log)
+		if err != nil {
+			return err
+		}
+	case "HelperLog":
+		err = cpLogFile(serverConfig.HelperLogFile, targetServerLogs, log)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// Collect daos metrics.
+func collectDaosMetrics(daosNodeLocation string, log logging.Logger, opts ...Params) error {
+	engineRunState, err := checkEngineState(log)
+	if err != nil {
+		return err
+	}
+
+	if engineRunState {
+		daos := copy{}
+		var cfgPath string
+		if opts[0].Config != "" {
+			cfgPath = opts[0].Config
+		} else {
+			cfgPath, _ = getServerConf(log)
+		}
+		serverConfig := config.DefaultServer()
+		serverConfig.SetPath(cfgPath)
+		serverConfig.Load()
+
+		for i := range serverConfig.Engines {
+			engineId := fmt.Sprintf("%d", i)
+			daos.Cmd = strings.Join([]string{"daos_metrics", "-S", engineId}, " ")
+
+			_, err := cpOutputToFile(daosNodeLocation, log, daos)
+			if err != nil {
+				return err
+			}
+		}
+	} else {
+		return errors.New("-- FAIL -- Daos Engine is not Running, so daos_metrics will not be collected")
+	}
+
+	return nil
+}
+
+// Collect output of system side option of daos_server command.
+func CollectDaosServerCmd(log logging.Logger, opts ...Params) error {
+	daosNodeLocation, err := createHostLogFolder(daosNodeLogFolder, log, opts...)
+	if err != nil {
+		return err
+	}
+
+	switch opts[0].LogCmd {
+	case "daos_metrics":
+		err = collectDaosMetrics(daosNodeLocation, log, opts...)
+		if err != nil {
+			return err
+		}
+	case "dump-topology":
+		hwlog := logging.NewCommandLineLogger()
+		hwProv := hwprov.DefaultTopologyProvider(hwlog)
+		topo, err := hwProv.GetTopology(context.Background())
+		if err != nil {
+			return err
+		}
+		f, err := os.Create(filepath.Join(daosNodeLocation, "daos_server_dump-topology"))
+		if err != nil {
+			return err
+		}
+		defer f.Close()
+		hardware.PrintTopology(topo, f)
+	}
+
+	return nil
+}
+
+// Common Entry/Exit point function.
+func CollectSupportLog(log logging.Logger, opts ...Params) error {
+	switch opts[0].LogFunction {
+	case "CopyServerConfig":
+		return CopyServerConfig(log, opts...)
```

Comment:
Do all of these functions need to be exported? Starting with a capital letter in Go means the symbol is exported. If they are solely called from this function I'd suggest unexporting them.